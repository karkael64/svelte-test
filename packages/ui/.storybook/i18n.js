import {
  register,
  init,
  getLocaleFromNavigator,
  locale as $locale,
} from "svelte-i18n";

const INIT_OPTIONS = {
  fallbackLocale: "en",
  initialLocale: null,
  loadingDelay: 200,
  formats: {},
  warnOnMissingMessages: true,
};

export function startClient() {
  register("en", () => import("../src/i18n/en.json"));
  register("fr", () => import("../src/i18n/fr.json"));

  init({
    ...INIT_OPTIONS,
    initialLocale: getLocaleFromNavigator(),
  });
}

export function setLocale(locale) {
  init(INIT_OPTIONS);
  $locale.set(locale);
}
