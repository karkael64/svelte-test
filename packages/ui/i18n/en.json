{
  "user-card": {
    "name": "Name",
    "email": "Email"
  }
}
