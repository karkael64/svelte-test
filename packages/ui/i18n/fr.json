{
  "user-card": {
    "name": "Nom",
    "email": "Courriel"
  }
}
