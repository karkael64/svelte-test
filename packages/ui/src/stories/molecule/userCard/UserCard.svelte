<script lang="ts">
  import { createEventDispatcher } from "svelte";
  import { t } from "svelte-i18n";

  type User = {
    name: string;
    email: string;
  };

  export let user: User;
  export let isEditing = false;

  const dispatch = createEventDispatcher();

  function onSubmit(event: Event) {
    dispatch("submit", event);
  }
</script>

{#if isEditing}
  <form on:submit|preventDefault={onSubmit}>
    <label class="label" name="name">
      <span class="title">{$t("user-card.name")}</span>
      <input type="text" class="input" value={user.name} />
    </label>
    <label class="label" name="email">
      <span class="title">{$t("user-card.email")}</span>
      <input type="text" class="input" value={user.email} />
    </label>
  </form>
{:else}
  <div>
    <div class="label" name="name">
      <span class="title">{$t("user-card.name")}</span>
      <span class="value">{user.name}</span>
    </div>
    <div class="label" name="email">
      <span class="title">{$t("user-card.email")}</span>
      <span class="value">{user.email}</span>
    </div>
  </div>
{/if}

<style>
  .label {
    display: flex;
  }

  .title {
    font-weight: 700;
  }

  .value {
    font-weight: 500;
  }
</style>
