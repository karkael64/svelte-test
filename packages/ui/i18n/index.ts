import { default as en } from "./en.json";
import { default as fr } from "./fr.json";

export const langs = {
  en,
  fr,
};
