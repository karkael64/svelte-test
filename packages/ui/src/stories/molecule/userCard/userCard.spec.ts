import UserCard, { User } from "./UserCard.svelte";
import { renderWithEvents } from "../../../../test/renderWithEvents";
import "../../../../test/i18n";

const commonUser: User = {
  name: "User",
  email: "user@email.com",
};

describe("UserCard", () => {
  it("Should instanciate with default values", async () => {
    const { getByText, container } = renderWithEvents(UserCard, {
      user: commonUser,
    });

    expect(getByText("Name")).toBeInTheDocument();
    expect(getByText("Email")).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });
});
