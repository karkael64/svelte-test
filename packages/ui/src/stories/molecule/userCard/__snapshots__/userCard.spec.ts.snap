// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserCard Should instanciate with default values 1`] = `
<body>
  <div>
    <div>
      <div
        class="label svelte-1d4gpnu"
        name="name"
      >
        <span
          class="title svelte-1d4gpnu"
        >
          Name
        </span>
         
        <span
          class="value svelte-1d4gpnu"
        >
          User
        </span>
      </div>
       
      <div
        class="label svelte-1d4gpnu"
        name="email"
      >
        <span
          class="title svelte-1d4gpnu"
        >
          Email
        </span>
         
        <span
          class="value svelte-1d4gpnu"
        >
          user@email.com
        </span>
      </div>
    </div>
    
  </div>
</body>
`;
